### PR TITLE
Lesson flow added

### PR DESCRIPTION
**Pull request by jannatulabeda**
Category, Lesson and Result view controller and storyboard added

### CODE REVIEW

**Comment by hasanmainul** [comment]
LGTM :+1:

File: fels_111/CategoryTableViewController.m [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+//  Copyright © 2016 Md Mainul Haque. All rights reserved.
+//
+
+#import "CategoryTableViewController.h"
+#import "CategoryTableViewCell.h"
+#import "LessonViewController.h"
+
+@interface CategoryTableViewController () <UITableViewDataSource, UITableViewDelegate>
+
+@property (strong, nonatomic) NSMutableArray *array;
```
Comment (TequilarTomsk) [inline fels_111/CategoryTableViewController.m]: Need to think more understandable name here

File: fels_111/CategoryTableViewController.m [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    // Uncomment the following line to preserve selection between presentations.
+    // self.clearsSelectionOnViewWillAppear = NO;
+    
+    // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
+    // self.navigationItem.rightBarButtonItem = self.editButtonItem;
+    
+    // Set title
+    self.title = @"Categories";
+    // Loading demo data
+    self.array = [NSMutableArray arrayWithObjects:@"Basic",@"Advance", @"Expert", nil];
```
Comment (TequilarTomsk) [inline fels_111/CategoryTableViewController.m]: I think Categories are loaded from api response?

File: fels_111/CategoryTableViewController.m [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    
+    CategoryTableViewCell *cell =(CategoryTableViewCell *) [tableView dequeueReusableCellWithIdentifier:@"categoryCell" forIndexPath:indexPath];
+    
+    // Configure the cell...
+    [cell.categoryImageView setImage: [UIImage imageNamed:@"avatar.png"]];
+    cell.categoryNameLabel.text = [self.array objectAtIndex:indexPath.row];
+    cell.learnedWordsLabel.text = @"You have learned 22/300";
```
Comment (TequilarTomsk) [inline fels_111/CategoryTableViewController.m]: is this only hard code for testing?

File: fels_111/CategoryTableViewController.m [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    cell.contentLabel.text = @"words, words, words, words, words........";
```
Comment (TequilarTomsk) [inline fels_111/CategoryTableViewController.m]: Same as above

File: fels_111/CategoryTableViewController.m [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    
+    return cell;
+}
+
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
```
Comment (TequilarTomsk) [inline fels_111/CategoryTableViewController.m]: why don't you use segue directly in storyboard?

File: fels_111/LessonViewController.m [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    [[self.answerButton2 layer] setBorderColor:[UIColor blackColor].CGColor];
+    
+    [[self.answerButton3 layer] setBorderWidth:1.0f];
+    [[self.answerButton3 layer] setBorderColor:[UIColor blackColor].CGColor];
+    
+    [[self.answerButton4 layer] setBorderWidth:1.0f];
+    [[self.answerButton4 layer] setBorderColor:[UIColor blackColor].CGColor];
+    
+    // Set done button
+    UIBarButtonItem * doneButton = [[UIBarButtonItem alloc]
```
Comment (TequilarTomsk) [inline fels_111/LessonViewController.m]: why dont you add this button and set UI for it in storyboard?

File: fels_111/LessonViewController.m [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+                                    initWithBarButtonSystemItem:UIBarButtonSystemItemDone
+                                    target:self
+                                    action:@selector(doneButtonPressed)];
+    self.navigationItem.rightBarButtonItem = doneButton;
+}
+
+// When done button is pressed
+-(void) doneButtonPressed {
+    
+    ResultTableViewController *resultTableViewController = [self.storyboard instantiateViewControllerWithIdentifier:@"ResultTableViewController"];
```
Comment (TequilarTomsk) [inline fels_111/LessonViewController.m]: You can use segue in storyboard for this

File: fels_111/ResultTableViewController.m [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    // Uncomment the following line to preserve selection between presentations.
+    // self.clearsSelectionOnViewWillAppear = NO;
+    
+    // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
+    // self.navigationItem.rightBarButtonItem = self.editButtonItem;
+    
+    self.array = [NSArray arrayWithObjects:@"ありがとう",@"ありがとう",@"ありがとう",@"ありがとう", nil];
```
Comment (TequilarTomsk) [inline fels_111/ResultTableViewController.m]: is this only hard code for tesing?

File: fels_111/ResultTableViewController.m [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+#import "ResultTableViewController.h"
+#import "ResultTableViewCell.h"
+
+@interface ResultTableViewController ()
+
+@property (weak, nonatomic) IBOutlet UILabel *lessonNameLabel;
+@property (weak, nonatomic) IBOutlet UILabel *learnedWordsLabel;
+
+@property (strong, nonatomic) NSArray *array;
```
Comment (TequilarTomsk) [inline fels_111/ResultTableViewController.m]: Need more understandable name

File: fels_111/ResultTableViewController.m [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    ResultTableViewCell *cell = (ResultTableViewCell *)[tableView dequeueReusableCellWithIdentifier:@"resultCell" forIndexPath:indexPath];
+    
+    // Configure the cell...
+    cell.resultImageView.image = [UIImage imageNamed:@"avatar.png"];
+    cell.contentLabel.text = [self.array objectAtIndex:indexPath.row];
+    cell.resultLabel.text = @"English";
```
Comment (TequilarTomsk) [inline fels_111/ResultTableViewController.m]: hard code??

File: fels_111/CategoryTableViewController.m (hunk 1 above)
Comment (jannatulabeda) [inline fels_111/CategoryTableViewController.m]: Just added for temporarily. Naming will be changed when other logics will be implemented.

File: fels_111/CategoryTableViewController.m (hunk 3 above)
Comment (jannatulabeda) [inline fels_111/CategoryTableViewController.m]: Yes everything is done only for testing. Sorry I should mention it in the comments or with the description of pull request.

File: fels_111/CategoryTableViewController.m (hunk 4 above)
Comment (jannatulabeda) [inline fels_111/CategoryTableViewController.m]: Everything is done only for testing. Sorry I should mention it in the comments or with the description of pull request.

File: fels_111/ResultTableViewController.m (hunk 10 above)
Comment (jannatulabeda) [inline fels_111/ResultTableViewController.m]: Just for testing.

File: fels_111/CategoryTableViewController.m (hunk 5 above)
Comment (jannatulabeda) [inline fels_111/CategoryTableViewController.m]: Definitely I will.

File: fels_111/LessonViewController.m (hunk 6 above)
Comment (jannatulabeda) [inline fels_111/LessonViewController.m]: Just added for temporarily.

File: fels_111/LessonViewController.m (hunk 7 above)
Comment (jannatulabeda) [inline fels_111/LessonViewController.m]: Okay I will use.

File: fels_111/ResultTableViewController.m (hunk 8 above)
Comment (jannatulabeda) [inline fels_111/ResultTableViewController.m]: Yes everything is just for testing purpose.

File: fels_111/ResultTableViewController.m (hunk 9 above)
Comment (jannatulabeda) [inline fels_111/ResultTableViewController.m]: Everything is done only for testing.

File: fels_111/ResultTableViewController.m (hunk 10 above)
Comment (jannatulabeda) [inline fels_111/ResultTableViewController.m]: Everything is done only for testing.

**Pull request merged**